tests/ballerina-test/src/test/java/org/ballerinalang/test/types/xml/XMLSecurityTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.test.types.xml;
+
+import org.ballerinalang.model.util.XMLUtils;
+import org.ballerinalang.model.values.BXML;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * Test cases for XML security.
+ */
+public class XMLSecurityTest {
+
+    @Test
+    public void testExternalEntityInjection() {
+        String xmlString = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" +
+                "<!DOCTYPE foo [" +
+                "<!ELEMENT foo ANY >" +
+                "<!ENTITY xxe SYSTEM \"https://www.w3schools.com/xml/note.xml\" >]>" +
+                "<foo>&xxe;</foo>";
+        BXML xmlDocument = XMLUtils.parse(xmlString);
+        Assert.assertEquals(xmlDocument.toString(), "<foo></foo>");
+    }
+
+    @Test (timeOut = 10000, expectedExceptions = org.ballerinalang.util.exceptions.BallerinaException.class)
```

Comment:
Don't we have to test the expected error message as well?

Reply:
Yes agreed. I wanted to do that too, but issue #4533 is blocking this. The current error message exposes Java related information and it is not the best thing to check on the current error here. WDYT?